backend/src/boundary_stats.rs

```diff
@@ -22,10 +22,14 @@ impl BoundaryStats {
         let mut simd = 0.0;
         if let Some(population_zones) = population_zones {
             for population_zone in population_zones {
-                if population_zone.geometry.intersects(polygon) {
-                    let overlap = polygon.intersection(&population_zone.geometry);
```

Comment:
re: https://github.com/a-b-street/ltn/pull/166#discussion_r1966478092

> is there lots of "wasted work" to first check for intersections, then calculate exactly what it is?

`intersects` is typically a little faster than `intersection`, so it's nice to do the quick path first.

I measured the difference in a temporary commit in the PR: https://github.com/a-b-street/ltn/pull/168/commits/29925abf75ac446b2aae6d1a7577b09511f8c13a

But more importantly, we can use a prepared geometry for `intersects` to speed it up even more (but we cannot do this with `intersection`).

For a little explanation, `intersects` is some topology code internal to geo, `intersection` is a wrapper around i_overlay.

Reply:
Thanks for the explanation!